### PR TITLE
Publish hotfix branches to docker.

### PR DESCRIPTION
**Pull request by kshakir**

### CODE REVIEW

File: src/bin/travis/afterSuccess.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+echo "BUILD_TYPE='$BUILD_TYPE'"
+echo "TRAVIS_BRANCH='$TRAVIS_BRANCH'"
+echo "TRAVIS_PULL_REQUEST='$TRAVIS_PULL_REQUEST'"
+
+if [ "$BUILD_TYPE" == "sbt" ] && [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
+
+    if [ "$TRAVIS_BRANCH" == "develop" ]; then
+        sbt 'set test in Test := {}' publish
+
+    elif [[ "$TRAVIS_BRANCH" =~ ^[0-9\.]+_hotfix$ ]]; then
```
Comment (kshakir) [inline src/bin/travis/afterSuccess.sh]: Only docker pushes hotfix branches. Do we have a tagging scheme for pre-release branches?

File: src/bin/travis/afterSuccess.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        docker login -u="$DOCKER_USERNAME" -p="$DOCKER_PASSWORD"
```
Comment (kshakir) [inline src/bin/travis/afterSuccess.sh]: Still awaiting these credentials to be added [to Travis](https://travis-ci.org/broadinstitute/cromwell/settings).

File: src/bin/travis/afterSuccess.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        sbt 'set test in Test := {}' -Dproject.isSnapshot=false dockerBuildAndPush
+
+    elif [[ "$TRAVIS_BRANCH" == "ks_docker_push" ]]; then
```
Comment (kshakir) [inline src/bin/travis/afterSuccess.sh]: DEBUG code to remove.

File: .travis.yml [hunk 4]
```diff
@@ -11,13 +11,13 @@ env:
   matrix:
     # Setting this variable twice will cause the 'script' section to run twice with the respective env var invoked
     - BUILD_TYPE=sbt
-    - BUILD_TYPE=checkPublish
-    - BUILD_TYPE=centaurJes
-    - BUILD_TYPE=centaurLocal
+    #- BUILD_TYPE=checkPublish
+    #- BUILD_TYPE=centaurJes
+    #- BUILD_TYPE=centaurLocal
```
Comment (kshakir) [inline .travis.yml]: DEBUG code to remove, makes the builds faster.

File: src/bin/travis/afterSuccess.sh (hunk 2 above)
Comment (geoffjentry) [inline src/bin/travis/afterSuccess.sh]: Tagging @hjfbynara

File: src/bin/travis/afterSuccess.sh (hunk 1 above)
Comment (geoffjentry) [inline src/bin/travis/afterSuccess.sh]: I think I don't understand the question. If I do I'd answer with the question "do we want to?"

File: src/bin/travis/afterSuccess.sh (hunk 2 above)
Comment (kshakir) [inline src/bin/travis/afterSuccess.sh]: Creds worked great for the test!

**Comment by kshakir** [comment]
Once this review is complete, I'll back port the patch to the actual hotfix branches `0.21_hotfix` and `0.22_hotfix`.

**Comment by ruchim** [comment]
@hjfbynara  would you mind being one of the two reviewers for this PR?

File: src/bin/travis/afterSuccess.sh (hunk 1 above)
Comment (kshakir) [inline src/bin/travis/afterSuccess.sh]: Discussed offline. Will only be pushing hotfix branches to docker, for now. Thus to kick off this process, the current operating procedure should be to create a new hotfix git branch whenever we release.

**Comment by hjfbynara** [comment]
Just wanted to confirm - what will the resulting tags look like?  

See that one will be tagged cromwellVersion.  And then there is another tagged with "version.value".  My scala is not all that great but will version.value result in cromwellVersion-7digitGITHASH ?

which is what this code does:

val basePrefix = baseVersion.map(_ + "-").getOrElse("")
def commitVersion: Option[String] = headCommit map (sha => basePrefix + sha.take(7))

**Comment by geoffjentry** [comment]
👍 provided that @hjfbynara is ok w/ the tag structure

**Comment by kshakir** [comment]
They'll look like the two tags currently in https://hub.docker.com/r/broadinstitute/cromwell/tags/ are:

- `23-0c25192`
- `23`

Note to future viewers: I'll be deleting the git hashed tag, but 23 should re-appear when 23 is actually released.

**Comment by hjfbynara** [comment]
👍

**Comment by ruchim** [comment]
👍 [proxy thumb for Henry]

**Pull request merged**